### PR TITLE
feat(EC-1950,EC-1982): test attestation parity and trust verification fixes

### PR DESCRIPTION
**Pull request by joejstuart**
## Summary

- **EC-1950**: Bring `test_attestation` package to feature parity with the `test` package — adds rules for warned results, informative tests, erred/skipped results, subject mismatch, unsupported result values, and rule data validation.
- **EC-1982**: Fix trust verification for in-toto test attestations attached as OCI referrers:
  - Fix `_has_trusted_provenance` to call `ec.sigstore.verify_attestation` directly on the referrer ref instead of looking for provenance via `ec.oci.image_referrers`. Tekton Chains stores SLSA provenance as cosign `.att` tags, not OCI referrers.
  - Fix `verified_statements` to use `parsed_blob_from_image` so the in-toto statement is extracted from the OCI artifact layer instead of returning the manifest JSON.

Jira: [EC-1950](https://redhat.atlassian.net/browse/EC-1950), [EC-1982](https://redhat.atlassian.net/browse/EC-1982)

## Test plan

- [x] All 998 ec-policies unit tests pass with 100% coverage
- [x] its-poc demo Phase 3: legitimate test attestation verified through Chains trust chain
- [x] its-poc demo: fake attestation correctly excluded (no Chains provenance)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by robnester-rh — DISMISSED** [review]

File: policy/release/test_attestation/test_attestation.rego [hunk 1]
```diff
@@ -96,7 +162,7 @@ deny contains result if {
 deny contains result if {
 	some statement in _test_attestations
 	statement.predicate.result
-	not statement.predicate.result in {"PASSED", "FAILED", "WARNED"}
+	not statement.predicate.result in {r | some r in rule_data.get("supported_test_attestation_results")}
```
Comment (simonbaird) [inline policy/release/test_attestation/test_attestation.rego]: I might be wrong, but is this just the same and shorter:

```
not statement.predicate.result in rule_data.get("supported_test_attestation_results")
```

Perhaps the motivation is to convert it to a set instead of a list, in which case I guess it's fine.

File: policy/release/test_attestation/test_attestation.rego [hunk 2]
```diff
@@ -131,28 +197,147 @@ deny contains result if {
...
 #   - redhat
 #   depends_on:
 #   - attestation_type.known_attestation_type
 #
-warn contains result if {
+deny contains result if {
 	some statement in _test_attestations
-	statement.predicate.result == "WARNED"
-	warned := _test_list(statement.predicate, "warnedTests")
+	statement.predicate.result in {r | some r in rule_data.get("erred_test_attestation_results")}
```
Comment (simonbaird) [inline policy/release/test_attestation/test_attestation.rego]: Same comment here.

File: policy/release/test_attestation/test_attestation.rego (hunk 1 above)
Comment (simonbaird) [reply policy/release/test_attestation/test_attestation.rego]: ..actually I can't see any value in converting to a set. IIUC "in \<array>" and "in \<set>" both do the same thing.

File: policy/release/test_attestation/test_attestation_test.rego [hunk 3]
```diff
@@ -567,83 +586,83 @@ test_missing_predicate if {
...
+	"failures": 0,
 })
 
-test_empty_failed_tests_array if {
+test_zero_failures_count if {
 	assertions.assert_equal_results(test_attestation.deny, {{
 		"code": "test_attestation.no_failed_tests",
-		"msg": "Test attestation \"empty-array-test\" has a failed result, failed tests (none listed)",
-		"term": "empty-array-test",
+		"msg": "Test attestation \"zero-count-test\" has a failed result, failures: 0",
```
Comment (simonbaird) [inline policy/release/test_attestation/test_attestation_test.rego]: I might be missing something, but I don't get this test. Is it a weird edge case? why would there be a deny with failure count zero? (If it it a weird edge case, maybe add a comment to explain that.)

File: policy/release/test_attestation/test_attestation_test.rego [hunk 4]
```diff
@@ -567,83 +586,83 @@ test_missing_predicate if {
...
+	"warnings": 0,
 })
 
-test_empty_warned_tests_array if {
+test_zero_warnings_count if {
 	assertions.assert_equal_results(test_attestation.warn, {{
 		"code": "test_attestation.no_test_warnings",
-		"msg": "Test attestation \"empty-warn-test\" has warnings, warned tests (none listed)",
-		"term": "empty-warn-test",
+		"msg": "Test attestation \"zero-warn-test\" has warnings, warnings: 0",
```
Comment (simonbaird) [inline policy/release/test_attestation/test_attestation_test.rego]: Same comment here.

File: policy/release/test_attestation/test_attestation_test.rego [hunk 5]
```diff
@@ -708,26 +727,28 @@ test_empty_string_result_value if {
...
-_mock_blob_non_string_tests(_) := _make_statement({
-	"result": "FAILED",
-	"configuration": [{"name": "bad-array-test"}],
-	"failedTests": ["CVE-2024-1234", 42, true],
+_mock_blob_count_triggers_deny(_) := _make_statement({
+	"result": "PASSED",
+	"configuration": [{"name": "count-trigger-test"}],
+	"successes": 0,
+	"failures": 5,
+	"warnings": 0,
```
Comment (simonbaird) [inline policy/release/test_attestation/test_attestation_test.rego:710]: Is it also an unlikely edge case? I'm happy to keep it if that is the case, but let's add a comment, since at first glance it seems weird and confusing.

File: antora/docs/modules/ROOT/pages/packages/release_test_attestation.adoc [hunk 6]
```diff
@@ -18,31 +18,31 @@ Produce a violation if any test result attestation has an erred result. The resu
 * Rule type: [rule-type-indicator failure]#FAILURE#
 * FAILURE message: `Test attestation %q has an erred result`
 * Code: `test_attestation.no_erred_test_attestations`
-* https://github.com/conforma/policy/blob/{page-origin-refhash}/policy/release/test_attestation/test_attestation.rego#L199[Source, window="_blank"]
+* https://github.com/conforma/policy/blob/{page-origin-refhash}/policy/release/test_attestation/test_attestation.rego#L208[Source, window="_blank"]
```
Comment (simonbaird) [inline antora/docs/modules/ROOT/pages/packages/release_test_attestation.adoc:21]: Nitpick: These docs fixups could be squashed into the commit where they belong. Not a blocker.

File: antora/docs/modules/ROOT/pages/packages/release_test_attestation.adoc [hunk 7]
```diff
@@ -90,7 +90,7 @@ Confirm the expected rule data keys have been provided in the expected format. T
 * Rule type: [rule-type-indicator failure]#FAILURE#
 * FAILURE message: `%s`
 * Code: `test_attestation.rule_data_provided`
-* https://github.com/conforma/policy/blob/{page-origin-refhash}/policy/release/test_attestation/test_attestation.rego#L294[Source, window="_blank"]
+* https://github.com/conforma/policy/blob/{page-origin-refhash}/policy/release/test_attestation/test_attestation.rego#L295[Source, window="_blank"]
```
Comment (simonbaird) [inline antora/docs/modules/ROOT/pages/packages/release_test_attestation.adoc]: Another docs fixup that could be squashed for tidiness. Again, just a nitpick, not a blocker.

File: policy/release/test_attestation/test_attestation.rego [hunk 8]
```diff
@@ -224,6 +224,7 @@ deny contains result if {
 #
 deny contains result if {
 	some statement in _test_attestations
+
```
Comment (simonbaird) [inline policy/release/test_attestation/test_attestation.rego:227]: Nitpick: Could also squash the fmt fixes into the commits where they belong. Again, not a blocker.

**Review by simonbaird — COMMENTED** [review]
A few suggestions to consider, otherwise lgtm.

**Review by simonbaird — DISMISSED** [review]

**Review by simonbaird — APPROVED** [review]
